qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -384,13 +384,12 @@ def _assemble(self, circuits, parameter_binds=None, **run_options):
             if parameter_binds:
                 # Handle parameter binding
                 parameterizations = self._convert_binds(circuits, parameter_binds)
-                assemble_binds = []
-                assemble_binds.append({param: 1 for bind in parameter_binds for param in bind})
+                circuits = [circuit.bind_parameters({param: 1 for param in circuit.parameters})
+                            for circuit in circuits]
```

Comment:
I'm worried about doing this, `bind_parameters()` adds a deepcopy for each circuit to bind the dummy value. The whole parameter binding path is setup to avoid doing the bind on the circuit object so we don't need to copy it. That being said I think there is a deepcopy already in `assemble()`: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/compiler/assembler.py#L553-L611 so there might not be any extra overhead here. But we should test it to confirm either way.

That being said I think we're seeing more evidence that aer should rewrite `assemble()` and not leverage terra's anymore. The serialization aer needs is more specific than what terra does now (this also gives us more flexibility once we have #1524 )

Reply:
The potential issue with that is doing it all at once is run in multiple processes. But I don't think that's a big loss so this is probably a fine approach.